ql/src/java/org/apache/hadoop/hive/ql/exec/DDLPlanUtils.java

```diff
@@ -691,20 +691,26 @@ public void getAlterTableStmtForeignKeyConstraint(ForeignKeyInfo fr, List<String
     }
     Map<String, List<ForeignKeyInfo.ForeignKeyCol>> all = fr.getForeignKeys();
     for (String key : all.keySet()) {
-      for (ForeignKeyInfo.ForeignKeyCol fkc : all.get(key)) {
-        ST command = new ST(ALTER_TABLE_ADD_FOREIGN_KEY);
-        command.add(CHILD_TABLE_NAME, fr.getChildTableName());
-        command.add(DATABASE_NAME, fr.getChildDatabaseName());
-        command.add(CONSTRAINT_NAME, key);
-        command.add(CHILD_COL_NAME, fkc.childColName);
-        command.add(DATABASE_NAME_FR, fkc.parentDatabaseName);
-        command.add(PARENT_TABLE_NAME, fkc.parentTableName);
-        command.add(PARENT_COL_NAME, fkc.parentColName);
-        command.add(ENABLE, fkc.enable);
-        command.add(VALIDATE, fkc.validate);
-        command.add(RELY, fkc.rely);
-        constraints.add(command.render());
+      List<ForeignKeyInfo.ForeignKeyCol> fkCols = all.get(key);
+      List<String> parentCols = new ArrayList<>();
+      List<String> childCols = new ArrayList<>();
```

Comment:
How about using `LinkedHashSet<String>(fkCols.size())` since column names should be unique.